internal/flags/flags.go

```diff
@@ -388,19 +390,45 @@ func GetSecretsFromFiles(rootCmd *cobra.Command) {
 
 // getSecretFromFile will check if the flag contains a reference to a file; if it does, replaces the value of the flag with the contents of the file.
 func getSecretFromFile(flags *pflag.FlagSet, secret string) {
-	value, err := flags.GetString(secret)
-	if err != nil {
-		log.Error(err)
-	}
-	if value != "" && isFile(value) {
-		file, err := ioutil.ReadFile(value)
-		if err != nil {
-			log.Fatal(err)
-		}
-		err = flags.Set(secret, strings.TrimSpace(string(file)))
+
+	flag := flags.Lookup(secret)
+	if flags.Lookup(secret).Value.Type() == "stringArray" {
+
+		values, err := flags.GetStringArray(secret)
 		if err != nil {
 			log.Error(err)
 		}
+
+		for _, value := range values {
+			if value != "" && isFile(value) {
+				file, err := ioutil.ReadFile(value)
+				if err != nil {
+					log.Fatal(err)
+				}
+
+				flag.Changed = false
+				rows := bytes.Split(file, []byte{'\n'})
+
+				for _, row := range rows {
+					err = flags.Set(secret, strings.TrimSpace(string(row)))
```

Comment:
Another issue: when using it as a file, you're only *adding* URLs, but you never remove the original value, aka the reference to the file. Causes the same error as above for me. Looks like this after the parsing is completed:
![image](https://user-images.githubusercontent.com/8885250/84566472-ad61d100-ad71-11ea-8b33-4ea3c59c2289.png)

Reply:
There is something weird going on with the flags API. It should clear it, and they are cleared in the tests, but I need to investigate this further.